keyboards/k320/k320.c

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2020 kuenhlee and Don Kjer
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef TKL_DURGOD_K320
+#error TKL_DURGOD_K320 not defined, recheck config.h
+#endif
+
+#include "k320.h"
+#include "bootloader_defs.h"
+#include "print.h"
+
+/* Private Functions */
+void off_all_leds(void) {
+    writePinHigh(LED_CAPS_LOCK);
+    writePinHigh(LED_SCROLL_LOCK);
+    writePinHigh(LED_WIN_LOCK);
+    writePinHigh(LED_MR_LOCK);
+}
+
+void on_all_leds(void) {
+    writePinLow(LED_CAPS_LOCK);
+    writePinLow(LED_SCROLL_LOCK);
+    writePinLow(LED_WIN_LOCK);
+    writePinLow(LED_MR_LOCK);
+}
+
+void bootmagic_lite(void) {
+    matrix_scan();
+    wait_ms(DEBOUNCE * 3u);
+    matrix_scan();
+
+    if (matrix_get_row(BOOTMAGIC_LITE_ROW) & (1u << BOOTMAGIC_LITE_COLUMN)) {
+        // Jump to bootloader.
+        bootloader_jump();
+    }
+}
+
+void matrix_init_kb(void) {
+    // put your keyboard start-up code here
+    // runs once when the firmware starts up
+    led_init_ports();
+    bootmagic_lite();
+    off_all_leds();
+    matrix_init_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    // put your per-action keyboard code here
+    // runs for every action, just before processing by the firmware
+#ifdef CONSOLE_ENABLE
+    dprintf("process_record_kb: %d, pressed: %d\n", keycode, record ? record->event.pressed : -1);
+#endif
+    return process_record_user(keycode, record);
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+
+    if (res) {
+        writePin(LED_CAPS_LOCK, !led_state.caps_lock);
+        writePin(LED_SCROLL_LOCK, !led_state.scroll_lock);
```

Comment:
These could be replaced with 
https://docs.qmk.fm/#/feature_led_indicators?id=configuration-options

Reply:
Updated to use config.h